assets/hip-1056/protobuf/stream/output/smart_contract_service.proto

```diff
@@ -129,33 +103,119 @@ message ContractSystemDeleteOutput {}
  * the original transaction.
  */
 message EthereumOutput {
+     /**
+      * A receipt for an Ethereum Transaction execution
+      */
+     EVMTransactionReceipt evm_transaction_receipt = 1;
+}
+
+/**
+ * EVM transaction execution log storage details
+ * Details maps to the log object in eth_getTransactionReceipt response without repeating info already available in the input transaction
+ */
+message EVMTransactionLog {
+     /**
+      * The address of the contract emitting the log
+      */
+     string address = 1;
+
+     /**
+      * The sequence index of the log event in this contract execution
+      */
+     int32 log_index = 2;
+
+     /**
+      * The Log data
+      */
+     bytes data = 3;
+
+     /**
+      * The logc topics
+      */
+     repeated bytes topics = 4;
+}
+
+/**
+ * EVM tranactin execution output details
+ * Details maps to the log eth_getTransactionReceipt response without repeating info already available in the input transaction
+ */
+message EVMTransactionOutput {
+     /**
+      * A bloom filter produced by this contract call.<br/>
+      * Ethereum uses this bloom filter to search for call results in the
+      * Ethereum block history. High false positive rates make the bloom
+      * filters quite limited value.
+      */
+     bytes bloom = 1;  
+
+     /**
+      * Result data from the function call.
+      * <p>
+      * This SHALL be encoded in RLP bytecode format.
+      */
+     bytes contractCallResult = 2;
+
+     /**
+      * Any error message produced by the contract call.
+      * <p>
+      * This SHALL be unset if the contract call succeeded.
+      */
+     string errorMessage = 3;
+
+     /**
+      * A quantity of "gas" used.<br/>
+      * This represents the resource units expended to execute this
+      * contract call, and correlates to transaction costs.
+      */
+     uint64 gas_used = 4;
+
+     /**
+      * Any Log events produced by this contract call.
+      */
+     repeated EVMTransactionLog logs = 5;
+}
+
+/**
+ * The slots read during the contract execution. These slots may or may not have been overwritten.
+ * The associated writes and contract will be found in the state changes MapUpdateChange KVs
+ * Q: Should we add the ContractId in here for simplicity?
+ */
+message SlotRead {
+     /**
+      * The contract storage slot counter used to correlate slots across BlockItems for a given contract in a transactions.
+      */
+     int32 index = 1;
```

Comment:
how would the index work?

My understanding is the index here needs to match the index in a `StorageSlot` in state changes, and then the value read for the identified slot is here and the value written to is in the corresponding `StorageSlot`.

What about storage slots only read during EVM execution?

Reply:
> My understanding is the index here needs to match the index in a StorageSlot in state changes, and then the value read for the identified slot is here and the value written to is in the corresponding StorageSlot.

Correct assumption.
So the hope is to replace the need for restating the storage slot address here. 
So that when a BSs parses sees a Contract-SlotAddress-Index item in the MapUpdateChange in the output then when they parse the TraceData here they can correlate this SlotRead to that item easily.
The index is a counter only for output and not in state. 
I need to confirm if the counter is across the single transaction or block. Also need to confirm there's an order already available to CN during processing or if this will be arbitrary at output time.

Reply:
> What about storage slots only read during EVM execution?

Hmmm, it should still be possible to have have the entry as the ContractID would still be known.
However, the key would be off so maybe index here should be a `oneof` for `index` or `key`.
If there's a partner in Changes then it's an index, if it's only read then it's the key.
Good catch

Reply:
yeah, agree we need a oneof and `index` is set if it's a slot written to, otherwise a value of type with both contract id and slot key need to be set in the oneof (a slot only read). We do need the contract id since there's no other place to get the info.

A size optimization we can consider, if it's a new storage slot, we can skip its slot read in the trace data, because the value is always an empty byte array. The saving is a `SlotRead` message with index set.

Reply:
> We do need the contract id since there's no other place to get the info.

Yeah within the context of other items the counter might have to be moved into actual state in which case it exists within the context of a contract, which means yeah a ContractId needs to be repeated again